source/ota_over_mqtt_demo.c

```diff
@@ -83,52 +83,91 @@
 /**
  * @brief The maximum size of the file paths used in the demo.
  */
-#define otaexampleMAX_FILE_PATH_SIZE      ( 260 )
+#define otaexampleMAX_FILE_PATH_SIZE                     ( 260 )
 
 /**
  * @brief The maximum size of the stream name required for downloading update file
  * from streaming service.
  */
-#define otaexampleMAX_STREAM_NAME_SIZE    ( 128 )
+#define otaexampleMAX_STREAM_NAME_SIZE                   ( 128 )
 
 /**
  * @brief The delay used in the OTA demo task to periodically output the OTA
  * statistics like number of packets received, dropped, processed and queued per connection.
  */
-#define otaexampleTASK_DELAY_MS           ( 1000U )
+#define otaexampleTASK_DELAY_MS                          ( 1000U )
 
 /**
  * @brief The maximum time for which OTA demo waits for an MQTT operation to be complete.
  * This involves receiving an acknowledgment for broker for SUBSCRIBE, UNSUBSCRIBE and non
  * QOS0 publishes.
  */
-#define otaexampleMQTT_TIMEOUT_MS         ( 5000U )
+#define otaexampleMQTT_TIMEOUT_MS                        ( 5000U )
 
+/**
+ * @brief The common prefix for all OTA topics.
+ *
+ * Thing name is substituted with a wildcard symbol `+`. OTA agent
+ * registers with MQTT broker with the thing name in the topic. This topic
+ * filter is used to match incoming packet received and route them to OTA.
+ * Thing name is not needed for this matching.
```

Comment:
The jobs docs say the client must confirm the thingname.  Does that happen in this code?
( https://docs.aws.amazon.com/iot/latest/developerguide/jobs-mqtt-api.html#jobs-mqtt-note )

Reply:
I will add the thing name comparison logic into the demo.

Reply:
Added thing name validation [here](https://github.com/FreeRTOS/Lab-Project-coreMQTT-Agent/pull/20/commits/1cbafc06422b8955b6398b2b7e6846fc1a0f21d9#diff-dca26c73ee244b83e7c4dbba30ffb673a144a4f937dc8403a25c60c6f1746aafR677)